src/app/functions/server/onerep.ts

```diff
@@ -117,12 +117,42 @@ async function onerepFetch(
   if (!onerepApiKey) {
     throw new Error("ONEREP_API_KEY env var not set");
   }
-  const url = new URL(path, onerepApiBase);
-  const headers = new Headers(options.headers);
-  headers.set("Authorization", `Bearer ${onerepApiKey}`);
-  headers.set("Accept", "application/json");
-  headers.set("Content-Type", "application/json");
-  return fetch(url, { ...options, headers });
+  try {
+    const url = `${onerepApiBase}${path}`;
+    const headers = new Headers(options.headers);
+    headers.set("Authorization", `Bearer ${onerepApiKey}`);
+    headers.set("Accept", "application/json");
+    headers.set("Content-Type", "application/json");
+
+    const response = await fetch(url, { ...options, headers });
+
+    if (!response.ok) {
+      const status = {
+        status: response.status,
+        statusText: response.statusText,
+      };
+      logger.error("onerep_response_not_ok", status);
+      throw new Error(`OneRep fetch failed: ${JSON.stringify(status)}`);
+    }
+
+    logger.info("onerep_fetch", { path, options });
+
+    return (await response.json()) as Promise<unknown>;
```

Comment:
I think (?) that TypeScript infers the `Promise` from it being an async function, so this annotation might result in a `Promise<Promise<unknown>>`?

```suggestion
    return (await response.json()) as unknown;
```